flink-contrib/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBMapState.java

```diff
@@ -0,0 +1,579 @@
+		}
+
+		try {
+			byte[] rawKeyBytes = serializeUserKeyWithCurrentKeyAndNamespace(userKey);
+			byte[] rawValueBytes = serializeUserValue(userValue);
+
+			backend.db.put(columnFamily, writeOptions, rawKeyBytes, rawValueBytes);
+		} catch (RocksDBException e) {
+			throw new RuntimeException("Error while putting data into RocksDB", e);
+		}
+	}
+
+	@Override
+	public void remove(UK userKey) throws IOException {
+		try {
+			byte[] rawKeyBytes = serializeUserKeyWithCurrentKeyAndNamespace(userKey);
+
+			backend.db.remove(columnFamily, writeOptions, rawKeyBytes);
+		} catch (RocksDBException e) {
+			throw new RuntimeException("Error while removing data from RocksDB.", e);
+		}
+	}
+
+	@Override
+	public boolean contains(UK userKey) throws IOException {
+		try {
+			byte[] rawKeyBytes = serializeUserKeyWithCurrentKeyAndNamespace(userKey);
+			byte[] rawValueBytes = backend.db.get(columnFamily, rawKeyBytes);
+
+			return (rawValueBytes != null);
+		} catch (RocksDBException e) {
+			throw new RuntimeException("Error while getting data from RocksDB", e);
+		}
+	}
+
+	@Override
+	public int size() throws IOException {
+		Iterator<Map.Entry<UK, UV>> iterator = iterator();
+
+		int count = 0;
+		while (iterator.hasNext()) {
+			count++;
+			iterator.next();
+		}
+
+		return count;
+	}
+
+	@Override
+	public Iterable<UK> keys() {
+		return new Iterable<UK>() {
+			@Override
+			public Iterator<UK> iterator() {
+				return new RocksDBMapIterator<UK>(backend.db, serializeCurrentKey()) {
+					@Override
+					public UK next() {
+						RocksDBMapEntry entry = nextEntry();
+						return (entry == null ? null : entry.getKey());
+					}
+				};
+			}
+		};
+	}
+
+	@Override
+	public Iterable<UV> values() {
+		return new Iterable<UV>() {
+			@Override
+			public Iterator<UV> iterator() {
+				return new RocksDBMapIterator<UV>(backend.db, serializeCurrentKey()) {
+					@Override
+					public UV next() {
+						RocksDBMapEntry entry = nextEntry();
+						return (entry == null ? null : entry.getValue());
+					}
+				};
+			}
+		};
+	}
+
+	@Override
+	public Iterator<Map.Entry<UK, UV>> iterator() {
+		return new RocksDBMapIterator<Map.Entry<UK, UV>>(backend.db, serializeCurrentKey()) {
+			@Override
+			public Map.Entry<UK, UV> next() {
+				return nextEntry();
+			}
+		};
+	}
+
+	@Override
+	public Iterable<Map.Entry<UK, UV>> get() throws Exception {
+		final Iterator<Map.Entry<UK, UV>> iterator = iterator();
+
+		// Return null to make the behavior consistent with other states.
+		if (!iterator.hasNext()) {
+			return null;
+		} else {
+			return new Iterable<Map.Entry<UK, UV>>() {
+				@Override
+				public Iterator<Map.Entry<UK, UV>> iterator() {
+					return iterator;
+				}
+			};
+		}
+	}
+
+	@Override
+	public void add(Map<UK, UV> map) throws Exception {
+		if (map == null) {
+			return;
+		}
+		
+		for (Map.Entry<UK, UV> entry : map.entrySet()) {
+			put(entry.getKey(), entry.getValue());
+		}
+	}
+	
+	@Override
+	public void clear() {
```

Comment:
I think that doesn't work be because the iterator doesn't iterator over the whole column family, just over the entries of the current state key, which can have several user-key/value pairs.